nifi-nar-bundles/nifi-social-media-bundle/nifi-twitter-processors/src/main/java/org/apache/nifi/processors/twitter/ConsumeTwitter.java

```diff
@@ -0,0 +1,366 @@
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.ProcessorInitializationContext;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.io.OutputStreamCallback;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Set;
+import java.util.List;
+import java.util.HashSet;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+@PrimaryNodeOnly
+@SupportsBatching
+@InputRequirement(Requirement.INPUT_FORBIDDEN)
+@Tags({"twitter", "tweets", "social media", "status", "json"})
+@CapabilityDescription("Streams tweets from Twitter's streaming API v2. The stream provides a sample stream or a search "
+    + "stream based on previously uploaded rules. This processor also provides a pass through for certain fields of the "
+    + "tweet to be returned as part of the response. See "
+    + "https://developer.twitter.com/en/docs/twitter-api/data-dictionary/introduction for more information regarding the "
+    + "Tweet object model. \n\n"
+    + "Warning: the underlying Java SDK used is still in beta as of the publishing of this processor feature.")
+@WritesAttribute(attribute = "mime.type", description = "Sets mime type to application/json")
+public class ConsumeTwitter extends AbstractProcessor {
+
+    static final AllowableValue ENDPOINT_SAMPLE = new AllowableValue(StreamEndpoint.SAMPLE_ENDPOINT.getEndpointName(),
+            "Sample Stream",
+            "Streams about one percent of all Tweets. " +
+                    "https://developer.twitter.com/en/docs/twitter-api/tweets/volume-streams/api-reference/get-tweets-sample-stream");
+    static final AllowableValue ENDPOINT_SEARCH = new AllowableValue(StreamEndpoint.SEARCH_ENDPOINT.getEndpointName(),
+            "Search Stream",
+            "The endpoint that provides a stream of tweets that matches the rules you added to the stream. " +
+                    "If rules are not configured, then the stream will be empty. " +
+                    "https://developer.twitter.com/en/docs/twitter-api/tweets/filtered-stream/api-reference/get-tweets-search-stream");
+
+    public static final PropertyDescriptor ENDPOINT = new PropertyDescriptor.Builder()
+            .name("stream-endpoint")
+            .displayName("Stream Endpoint")
+            .description("Specifies which endpoint tweets should be streamed. " +
+                    "Usage of search endpoint requires that rules be uploaded beforehand. See " +
+                    "https://developer.twitter.com/en/docs/twitter-api/tweets/filtered-stream/api-reference/" +
+                    "post-tweets-search-stream-rules")
```

Comment:
This can be simplified since the individual values describe what is necessary for each option.
```suggestion
            .description("The source from which the processor will consume Tweets.")
```

Reply:
Makes sense, making the changes